src/components/templates/ScreeningRoom/ScreeningRoom.tsx

```diff
@@ -0,0 +1,153 @@
+import React, { useMemo, useCallback, useState } from "react";
+import { faSearch, faTimesCircle } from "@fortawesome/free-solid-svg-icons";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+
+import { IFRAME_ALLOW } from "settings";
+
+import { GenericVenue } from "types/venues";
+import { ScreeningRoomVideo } from "types/screeningRoom";
+
+import { WithId } from "utils/id";
+
+import { useScreeningRoom } from "./useScreeningRoom";
+
+import { InputField } from "components/atoms/InputField";
+import { Button } from "components/atoms/Button";
+import { PosterCategory } from "components/atoms/PosterCategory";
+
+import { ScreeningVideoPreview } from "./components/ScreeningVideoPreview";
+
+import "./ScreeningRoom.scss";
+
+export interface ScreeningRoomProps {
+  venue: WithId<GenericVenue>;
+}
+
+export const ScreeningRoom: React.FC<ScreeningRoomProps> = ({ venue }) => {
+  const {
+    videos,
+    isVideosLoaded,
+
+    increaseDisplayedVideosAmount,
+
+    categoryList,
+    subCategoryList,
+
+    searchInputValue,
+    categoryFilter,
+    subCategoryFilter,
+
+    setSearchInputValue,
+    setCategoryFilter,
+    setSubCategoryFilter,
+  } = useScreeningRoom(venue.id);
+
+  const [selectedVideo, setSelectedVideo] = useState<
+    WithId<ScreeningRoomVideo>
+  >();
+
+  const selectedVideoId = selectedVideo?.id;
+
+  const unSelectVideo = useCallback(() => setSelectedVideo(undefined), []);
+
+  const onInputFieldChange = useCallback(
+    (e: React.ChangeEvent<HTMLInputElement>) => {
+      setSearchInputValue(e.target.value);
+    },
+    [setSearchInputValue]
+  );
+
+  const renderedVideoPreviews = useMemo(
+    () =>
+      videos.map((video) => (
+        <ScreeningVideoPreview
+          key={video.id}
+          video={video}
+          selectThisVideo={() => setSelectedVideo(video)}
+          selected={video.id === selectedVideoId}
+        />
+      )),
+    [videos, selectedVideoId]
+  );
+
+  const renderedCategoryOptions = useMemo(() => {
+    return (
+      <div className="ScreeningRoom__categories">
+        {categoryList.map((category) => (
+          <PosterCategory
+            key={category}
+            category={category}
+            onClick={() => setCategoryFilter(category)}
+            containerClassname="ScreeningRoom__category"
+            active={category === categoryFilter}
+          />
+        ))}
+      </div>
+    );
+  }, [categoryList, categoryFilter, setCategoryFilter]);
+
+  const renderedSubCategoryOptions = useMemo(
+    () => (
+      <div className="ScreeningRoom__subcategories">
+        {subCategoryList.map((subCategory) => (
+          <PosterCategory
+            key={subCategory}
+            category={subCategory}
+            onClick={() => setSubCategoryFilter(subCategory)}
```

Comment:
Same here.

Reply:
It's useMemoed
![image](https://user-images.githubusercontent.com/63194656/119567744-5d7aef00-bdb5-11eb-889f-08fabb6884d8.png)